### PR TITLE
feat(io-engine/cores): add missing core list

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Allows specifying the exact list of cores.
Examples:
--set='io_engine.coreList=1'
--set='io_engine.coreList={1,2,3}'

Also adds validation to ensure the coreCount is always >= 1 and all cores are unique.

### CODE REVIEW

**Review by sinhaashish — APPROVED** [review]

File: chart/values.yaml [hunk 1]
```diff
@@ -273,8 +273,10 @@ io_engine:
   envcontext: ""
   reactorFreezeDetection:
     enabled: false
-  # -- The number of cpu that each io-engine instance will bind to.
+  # -- The number of cores that each io-engine instance will bind to.
   cpuCount: "2"
+  # -- Overrides the cpuCount and explicitly sets the list of cores.
+  coreList: []
```
Comment (avishnu) [inline chart/values.yaml:279]: I assume the override happens only if the list is non-empty, right?

**Review by avishnu — APPROVED** [review]
lgtm

File: chart/values.yaml [hunk 2]
```diff
@@ -260,7 +260,7 @@ csi:
 
 io_engine:
   # -- Log level for the io-engine service
-  logLevel: info,io_engine=info
```
Comment (niladrih) [inline chart/values.yaml:263]: This change requires additional instructions for upgrade, Is this necessary for the 2.0 branch?

**Review by niladrih — APPROVED** [review]
LGTM, few nitpick comments.

File: chart/values.yaml (hunk 1 above)
Comment (tiagolobocastro) [reply chart/values.yaml:279]: Yes correct

[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply chart/values.yaml:279]: Yeah I can add an example. What do you mean by an atypical option?

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**